specification/cognitiveservices/OpenAI.Inference/models/completions.create.cadl

```diff
@@ -150,12 +150,14 @@ model Choice {
     finish_reason?: string;
 }
 
+alias NullableFloat = float32 | null;
+
 @doc("LogProbs model within completion choice")
 model CompletionsLogProbs {
     @doc("Tokens")
     tokens?: string[];
     @doc("LogProbs of Tokens")
-    token_logprobs?: float32[];
+    token_logprobs?: NullableFloat[];
```

Comment:
Does `token_logprobs: null` have some special significance that is different from a `CompletionLogProbs` that does not contain a `token_logprobs`property?

Reply:
@trrwilson will have more details but when asking for an echo in the request you get an empty entry in this array in order to maintain index synchronization between all of the arrays that come back.

This is different than the entire property is not there as that will simply have an empty array of floats, however in this case the array is not empty but can contain [ null, 0.1, 2.4, ... ].  [Here is a recording](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/openai/Azure.AI.OpenAI/tests/SessionRecords/OpenAIInferenceTests/TestLogProbs.json#L71) of what the service does.

Reply:
Very late to respond, but Michael's completely right: for whatever reason, both Azure OpenAI and its non-Azure counterpart return a "null" probability entry as the first item in the float and object arrays when `echo` (include the prompt in the generated completion) is enabled and those probabilities are requested. I can't speak to the intended purpose or value of having that null, nor can I even intimate a reason that "echo" is inherently valuable on the API surface, but pending strong justification to remove it and subset OpenAI's parameterization on the SDK (which I'd like, but is hard), we'll have this situation pop up where nullable floats can appear in some responses.